plugins/Geant4EventReaderGuineaPig.cpp

```diff
@@ -91,19 +92,32 @@ Geant4EventReaderGuineaPig::~Geant4EventReaderGuineaPig()    {
   m_input.close();
 }
 
+
 Geant4EventReader::EventReaderStatus
-Geant4EventReaderGuineaPig::moveToEvent(int event_number) {
+Geant4EventReaderGuineaPig::setParameters( std::map< std::string, std::string > & parameters ) {
+
+  _getParameterValue( parameters, "ParticlesPerEvent", m_part_num, -1);
   
-  //  if( m_currEvent == 0 && event_number > 0 ){
-  if( m_part_num < 0 ){
+  if( m_part_num <  0 ) 
+    printout(INFO,"EventReaderGuineaPig::setParameters"," --- Will read all particles in pairs file into one event " );
+  else
+    printout(INFO,"EventReaderGuineaPig::setParameters"," --- Will read %d particles per event from pairs file ", m_part_num );
+    
+  return EVENT_READER_OK;
+}
 
-    m_part_num = event_number ;
+Geant4EventReader::EventReaderStatus
+Geant4EventReaderGuineaPig::moveToEvent(int event_number) {
+  
+  printout(DEBUG,"EventReaderGuineaPig::moveToEvent"," event_number: %d , m_currEvent %d",
+	   event_number,m_currEvent ) ;
   
-    printout(INFO,"EventReaderGuineaPig::moveToEvent"," --- Will read %d particles per event from pairs file ", m_part_num );
-    printout(INFO,"EventReaderGuineaPig::moveToEvent"," ***  Due to a workaround event numbers will start from %d  for now !!", m_part_num );
+  if( m_currEvent == 0 && event_number > 0 ){
 
+    printout(INFO,"EventReaderGuineaPig::moveToEvent"," --- Cannot skip to event %d in GuineaPig files - ignored ! ", event_number ); 
```

Comment:
Yes, skipping over the N particles is what should be done, as was possible and very useful in Mokka.
It allows one to split the simulation of one pair background file into multiple jobs, or skip to the one particle that crashes things.